test/model/message_list_test.dart

```diff
@@ -1704,6 +1704,25 @@ void main() {
       }
     });
   });
+  group('topics compared case-insensitively', () {
+    void doTest(String description, String topicA, String topicB, bool expected) {
+      test(description, () {
+        final stream = eg.stream();
+        final messageA = eg.streamMessage(stream: stream, topic: topicA);
+        final messageB = eg.streamMessage(stream: stream, topic: topicB);
+        check(haveSameRecipient(messageA, messageB)).equals(expected);
+      });
+    }
+
+    doTest('same case, all lower',               'abc',  'abc',  true);
+    doTest('same case, all upper',               'ABC',  'ABC',  true);
+    doTest('same case, mixed',                   'AbC',  'AbC',  true);
+    doTest('same non-letter chars',              '嗎',    '嗎',    true);
```

Comment:
```suggestion
    doTest('same non-cased chars',               '嗎',    '嗎',    true);
```

Calling this character "non-letter" is confusing, because Unicode in fact classifies it in its "Letter" category, specifically "Lo (Letter, Other)":
```
$ unicode --long 嗎
U+55CE CJK UNIFIED IDEOGRAPH-55CE
UTF-8: e5 97 8e UTF-16BE: 55ce Decimal: &#21966; Octal: \052716
嗎
Category: Lo (Letter, Other); East Asian width: W (wide)
Unicode block: 4E00..9FFF; CJK Unified Ideographs
Bidi: L (Left-to-Right)
```

(By comparison, cased letters like `a` and `A` are in categories "Ll (Letter, Lowercase)" and "Lu (Letter, Uppercase)" respectively.)